### ISSUE TITLE
AMD CPU and v2.3

### ISSUE THREAD

**Issue by sunnsurf** [Microsoft/CNTK#2793]
After updating (via NuGet) to 2.3 the cross-entropy with softmax (XE) and classification error (CE) calculations seem incorrect - the XE is returning a value much lower than expected and CE is returning 0, from the first epoch.

It potentially relates to AMD CPU as the above behaviour is observed independent of device selection (CPU or GPU). Additionally, the exact same code has been run on a laptop (non-Nvidia GPU) and Azure NC VM (Tesla K80 GPU) and both calculate more reasonable XE and CE values.

**Comment by ebarsoumMS** [comment]
Does it work fine with 2.2?

**Comment by sunnsurf** [comment]
Feeling a little confused and daft now. After downgrading to 2.2 to capture run results then upgrading again, calculations appear to be working (2.2, 2.3 & 2.3.1), that is, returning non-0 result.

**Comment by ebarsoumMS** [comment]
Please activate, if it repro again.
